src/diff_parse.c

```diff
@@ -44,7 +44,11 @@ static git_diff_parsed *diff_parsed_alloc(void)
 	diff->base.patch_fn = git_patch_parsed_from_diff;
 	diff->base.free_fn = diff_parsed_free;
 
-	git_diff_init_options(&diff->base.opts, GIT_DIFF_OPTIONS_VERSION);
+	if (git_diff_init_options(&diff->base.opts, GIT_DIFF_OPTIONS_VERSION) < 0) {
```

Comment:
I was surprised to find that the `git_*_init_options` functions return an `int`, since none of them can fail.  I guess it's nice future proofing.  Anyway, if this shuts coverity up then we should take it.